File: CS-Examples/07_DocumentOperation/KeepSameFormat.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;

namespace KeepSameFormat
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new instance of the Document class and load a document from the specified file path ("Template_N2.docx").
			Document srcDoc = new Document();
			srcDoc.LoadFromFile(@"..\..\..\..\..\..\..\Data\Template_N2.docx");

			// Create a new instance of the Document class and load another document from the specified file path ("Template_N3.docx").
			Document destDoc = new Document();
			destDoc.LoadFromFile(@"..\..\..\..\..\..\..\Data\Template_N3.docx");

			// Set the KeepSameFormat property of the source document to true.
			srcDoc.KeepSameFormat = true;

			// Iterate through each section in the source document.
			foreach (Section section in srcDoc.Sections)
			{
				// Clone each section and add it to the destination document.
				destDoc.Sections.Add(section.Clone());
			}

			// Specify the output file name.
			string output = "KeepSameFormating_out.docx";

			// Save the modified destination document to a file with the specified output file name and format (Docx2013).
			destDoc.SaveToFile(output, FileFormat.Docx2013);

			// Clean up resources used by the source document.
			srcDoc.Dispose();

			// Clean up resources used by the destination document.
			destDoc.Dispose();

            //Launch the file
            WordDocViewer(output);
        }

        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB-Examples/07_DocumentOperation/KeepSameFormat.vb
Imports Spire.Doc
Imports Spire.Doc.Documents

Namespace KeepSameFormat
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new Document object for the source document
			Dim srcDoc As New Document()
			' Load the source document from the specified file path
			srcDoc.LoadFromFile("..\..\..\..\..\..\..\Data\Template_N2.docx")

			' Create a new Document object for the destination document
			Dim destDoc As New Document()
			' Load the destination document from the specified file path
			destDoc.LoadFromFile("..\..\..\..\..\..\..\Data\Template_N3.docx")

			' Set the KeepSameFormat property of the source document to true
			srcDoc.KeepSameFormat = True

			' Iterate through each Section in the source document
			For Each section As Section In srcDoc.Sections
				' Clone and add each Section to the destination document
				destDoc.Sections.Add(section.Clone())
			Next section

			' Specify the output file name and path
			Dim output As String = "KeepSameFormating_out.docx"
			' Save the destination document to the specified output file
			destDoc.SaveToFile(output, FileFormat.Docx2013)

			' Dispose the source document object
			srcDoc.Dispose()
			' Dispose the destination document object
			destDoc.Dispose()

			'Launch the file
			WordDocViewer(output)
		End Sub

		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace
